flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/rules/logical/PushProjectIntoTableSourceScanRuleTest.java

```diff
@@ -352,8 +358,25 @@ public void testProjectionIncludingOnlyMetadata() {
                 .containsExactly("metadata");
     }
 
+    private void replaceProgramWithProjectMergeRule() {
+        FlinkChainedProgram programs = new FlinkChainedProgram<BatchOptimizeContext>();
+        programs.addLast(
+                "rules",
+                FlinkHepRuleSetProgramBuilder.<BatchOptimizeContext>newBuilder()
+                        .setHepRulesExecutionType(HEP_RULES_EXECUTION_TYPE.RULE_SEQUENCE())
+                        .setHepMatchOrder(HepMatchOrder.BOTTOM_UP)
+                        .add(
+                                RuleSets.ofList(
+                                        CoreRules.PROJECT_MERGE,
+                                        PushProjectIntoTableSourceScanRule.INSTANCE))
+                        .build());
+        util().replaceBatchProgram(programs);
+    }
```

Comment:
Why do we need to add this method in the pre-step commit？ Maybe it need to move to the Step2.